python/ray/data/grouped_data.py

```diff
@@ -107,6 +107,22 @@ class PushBasedGroupbyOp(_GroupbyOp, PushBasedShufflePlan):
     pass
 
 
+class _MultiColumnSortedKey:
+    """Acting as a tuple of group keys for np.searchsorted"""
+
+    __slots__ = ("data",)
```

Comment:
OOC what's the primary motivation for defining `__slots__` here?

Reply:
https://wiki.python.org/moin/UsingSlots
My understanding is that it speeds up creation and saves memory. I did see some speed improvement when creating a large amount of small object like this.

Reply:
Ah, yeah. Was wondering if there was any reason other than performance.

Sg.